src/sentry/integrations/slack/notifications.py

```diff
@@ -43,15 +43,23 @@ def get_channel_and_integration_by_user(
         # recipients.
         return None, None
 
+    identity = [
+        identity
+        for identity in identities
+        if identity.user.org_memberships.filter(id=organization.id)
+    ]
+    if not identity:
+        return None, None
+
     try:
         integration = Integration.objects.get(
-            provider=identity.idp.type,
+            provider=identity[0].idp.type,
```

Comment:
This will fix it, but we'll just choose a random identity out of the registered ones.